opal/mca/btl/vader/btl_vader.h

```diff
@@ -84,7 +84,11 @@ union vader_modex_t {
         void *segment_base;
     } xpmem;
 #endif
-    opal_shmem_ds_t seg_ds;
+    struct vader_modex_other_t {
+        int seg_ds_size;
+        opal_shmem_ds_t seg_ds;
```

Comment:
The seg_ds should always be last. We don't modex send the entire structure as it is padded out with 0's.

Reply:
okay, thanks, changed in the upcoming force push